backend/endpoints/scan.py

```diff
@@ -49,17 +50,21 @@ async def scan(_sid: str, platforms: str, complete_rescan: bool = True, sm = Non
             rom_id: int = dbh.rom_exists(scanned_platform.slug, rom["file_name"])
             if rom_id and not complete_rescan:
                 continue
-            await sm.emit("scan:scanning_rom", rom["file_name"])
-            await sm.emit("")  # Workaround to emit in real-time
-            log.info(f"\t - {rom['file_name']}")
-            if rom["multi"]:
-                [
-                    log.info(f"\t\t · {file}")
-                    for file in rom["files"]
-                ]
+
             scanned_rom: Rom = fastapi.scan_rom(scanned_platform, rom)
+            await sm.emit(
+                "scan:scanning_rom",
+                {
+                    "p_slug": scanned_platform.slug,
+                    "file_name": scanned_rom.file_name,
+                    "r_name": scanned_rom.r_name,
+                },
+                ignore_queue=True,
```

Comment:
By adding `ignore_queue=True` I'm able to get the socket to emit in real-time, so hopefully this negates the need for `sm.emit("")`.

Reply:
My god dude, was triying to "fix" this workaround a lot but never found the solution. This seems to be the perfect one